frontend/src/components/dashboard/campaigns/announcement-modal/AnnouncementModal.tsx

```diff
@@ -33,23 +40,84 @@ const AnnouncementModal = () => {
     }
   }
 
+  const mediaUrl = i18n._(ANNOUNCEMENT.mediaUrl)
+  let mediaAndTitle = null
+  if (isVideoUrl(mediaUrl)) {
+    mediaAndTitle = (
+      <>
+        <h4 className={`${styles.title} ${styles.titleTop}`}>
+          {i18n._(ANNOUNCEMENT.title)}
+        </h4>
+        <ReactPlayer
+          url={mediaUrl}
+          className={styles.modalMedia}
+          controls
+          playing
+          width="100%"
+          height="100%"
+        />
+      </>
+    )
+  } else {
+    mediaAndTitle = (
+      <>
+        <img
+          className={`${styles.modalMedia} ${styles.modalImg}`}
+          src={mediaUrl}
+          alt="Modal graphic"
+        ></img>
+        <h4 className={`${styles.title} ${styles.titleCentered}`}>
+          {i18n._(ANNOUNCEMENT.title)}
+        </h4>
+      </>
+    )
+  }
+
+  // In lingui, we can't simply leave a translation empty as a default will be used during compilation.
+  // Instead, mark it as "null" to indicate that it is empty.
+  const EMPTY_TRANSLATION = 'null'
+  const subtextText = i18n._(ANNOUNCEMENT.subtext)
```

Comment:
How about using https://lingui.js.org/ref/core.html#options.missing to allow for empty values? 

```js
setupI18n({ catalogs, language, missing: () => '' })
```

Reply:
I tried that, but it appears that the `missing` is only called when a translation is missing from the compiled `messages.js` file.

However, whenever we run `npm run compile`, any missing translations in `messages.po` is replaced by the translation ID in `messages.js`. As a result, there will never be any translations actually missing in the `messages.js` file.

The same issue is explained here: https://github.com/lingui/js-lingui/issues/475#issue-422639166

Reply:
Thanks for explaining this!